python/pyarrow/io.pxi

```diff
@@ -1964,7 +1980,9 @@ cdef class Codec(_Weakrefable):
     @property
     def compression_level(self):
         """Returns the compression level parameter of the codec"""
-        return frombytes(self.unwrap().compression_level())
+        if self.name == 'snappy':
+            return None
+        return self.unwrap().compression_level()
```

Comment:
Should add a test for this? (I assume this was raising for Snappy?)

Reply:
Good point, it was failing as-is, `compression_level() -> int` and `frombytes` would fail trying to decode an int. Also modified snappy variant as that has no compression level and would invalid integer.